providers/goipmi/goipmi.go

```diff
@@ -0,0 +1,156 @@
+package goipmi
+
+import (
+	"context"
+	"errors"
+
+	"github.com/bmc-toolbox/bmclib/providers"
+	"github.com/gebn/bmc"
+	"github.com/gebn/bmc/pkg/ipmi"
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the provider implementation
+	ProviderName = "goipmi"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "ipmi"
+)
+
+var (
+	// Features implemented by goipmi
+	Features = registrar.Features{
+		providers.FeaturePowerSet,
+		providers.FeaturePowerState,
+	}
+)
+
+type Conn struct {
+	Host      string
+	Port      string
+	User      string
+	Pass      string
+	Log       logr.Logger
+	transport bmc.SessionlessTransport
+	conn      bmc.Session
+}
+
+func (c *Conn) Open(ctx context.Context) error {
+	machine, err := bmc.Dial(ctx, c.Host)
+	if err != nil {
+		return err
+	}
+	c.transport = machine
+
+	sess, err := machine.NewSession(ctx, &bmc.SessionOpts{
+		Username:          c.User,
+		Password:          []byte(c.Pass),
+		MaxPrivilegeLevel: ipmi.PrivilegeLevelOperator,
```

Comment:
Any reason we're going with the Operator priviledge here instead of [Administrator](https://github.com/gebn/bmc/blob/master/pkg/ipmi/privilege_level.go#L34)

Reply:
ah good catch, will update

Reply:
updated and pushed